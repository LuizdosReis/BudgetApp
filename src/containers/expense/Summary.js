import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../../selectors/expenses';
import getTotalExpenses from '../../selectors/expensesTotal';

export const Summary = ({ count, total }) => {
  const expenseWord = count === 1 ? 'despesa' : 'despesas';
  return (
    <div>
      <h1>
        {count}
        {' '}
        {expenseWord}
        {' '}
| total
        {' '}
        {numeral(total / 100).format('$0,0.00')}
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const expenses = getVisibleExpenses(state.expenses, state.expensesFilter);
  return {
    count: expenses.length,
    total: getTotalExpenses(expenses),
  };
};

export default connect(
  mapStateToProps,
  undefined,
)(Summary);
