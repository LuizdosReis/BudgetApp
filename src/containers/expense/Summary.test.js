import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Summary } from './Summary';

test('should correctly render summary with one expense', () => {
  const wrapper = shallow(<Summary count={1} total={235} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('should correctly render summary with  expense', () => {
  const wrapper = shallow(<Summary count={23} total={2353493029} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
