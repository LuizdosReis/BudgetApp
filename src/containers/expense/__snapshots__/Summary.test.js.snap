// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render summary with  expense 1`] = `
<div>
  <h1>
    23
     
    despesas
     
    | total
     
    $23,534,930.29
  </h1>
</div>
`;

exports[`should correctly render summary with one expense 1`] = `
<div>
  <h1>
    1
     
    despesa
     
    | total
     
    $2.35
  </h1>
</div>
`;
